templates/collections/blocks/promoted_pages.html

```diff
@@ -0,0 +1,44 @@
+{% load wagtailimages_tags %}
+
+<h2>{{ value.heading }}</h2>
+<p>{{ value.sub_heading }}</p>
+
+<ul class="row card-group">
+    {% for promoted_item in value.promoted_items %}
+
+        {% image promoted_item.teaser_image fill-288x172 as teaser_image_small %}
+        {% image promoted_item.teaser_image fill-328x196 as teaser_image_medium %}
+        {% image promoted_item.teaser_image fill-348x208 as teaser_image_large %}
+        {% image promoted_item.teaser_image fill-543x325 as teaser_image_extra_large %}
+
+        {% comment %}
+        Source copied from templates/includes/card-group-secondary-nav.html
+
+        Modified slightly to support data from a streamfield.
+
+        This is not intended to be production-ready code.
+
+        To discuss an approach to allow for reusability going forward.
+        {% endcomment %}
```

Comment:
Hi @danbentley - I think this is fine for the time-being. I'd rather have some pragmatic redundancy than additional complexity and the key thing is that we can determine where any given template is coming from, which we can from the block.